feat(ipam): add doc for new IPAM product (#3287)
scaleway/docs-content: network/ipam/concepts.mdx

The public rollout of IPAM functionality is being carried out gradually: IPAM is currently in [public beta](https://www.scaleway.com/en/betas/).

- In the [console](https://console.scaleway.com/ipam/), you can currently use IPAM to view, tag and filter all your [private IP addresses](#private-ip-address)
- With the [API](https://www.scaleway.com/en/developers/api/ipam/), you can additionally [reserve](https://www.scaleway.com/en/developers/api/ipam/#path-ips-book-a-new-ip) an available IP from a Private Network's CIDR block, or [release](https://www.scaleway.com/en/developers/api/ipam/#path-ips-release-an-ip) a previously-reserved address that you no longer want.

More functionality is planned for the future, including:

- Reserve/release private IP addresses in the console
- Select a reserved private IP address to use when attaching a resource to a Private Network
- Management of public IP addresses

## IPv4

> Internet Protocol Version 4 is the standard protocol used for IP addresses, and routes most internet traffic as of today. Each IPv4 address has 32 bits. Written in human-readable form, an IPv4 address is generally shown as four octets separated by periods, e.g. `72.16.254.1`.

Reviewer: The IP `72.16.254.1` belongs to the IP space of Microsoft. 
Any particular reason for this example?

Author: This has been in the doc for ages, I copied/pasted it into IPAM from elsewhere in the network doc. No particular reason as far as I know, we can change it to something SCW-ish

> Internet Protocol Version 4 is the standard protocol used for IP addresses, and routes most internet traffic as of today. Each IPv4 address has 32 bits. Written in human-readable form, an IPv4 address is generally shown as four octets separated by periods, e.g. `72.16.254.1`.

Author: ```suggestion
Internet Protocol Version 4 is the standard protocol used for IP addresses, and routes most internet traffic as of today. Each IPv4 address has 32 bits. Written in human-readable form, an IPv4 address is generally shown as four octets separated by periods, e.g. `151.115.59.87`.
```

## IPv6

Internet Protocol Version 6 is the most recent version of the IP protocol used for IP addresses. Each IPv6 address has 128 bits. Written in human-readable form, an IPv6 address can be shown as eight groups of four hexadecimal digits, each group representing 16 bits and separated by a colon, e.g. `2001:0DB8:0000:0003:0000:01FF:0000:002E`. This can also be notated as `2001:DB8::3:0:1FF:0:2E`.

IPv6-compatible Scaleway resources may have a [public](#public-flexible-ip-address) IPv6 address (flexible IP) for accessibility from the public internet, and will have a [private](#private-ip-address) IPv6 address for each Private Network they are attached to. 

## Private IP address

Private IP addresses identify devices on Private Networks. They are not routed on the public internet, and can be used to route traffic privately between resources attached to the Private Network, as well as to [other Private Networks in the same VPC](/network/vpc/concepts/#routing).

> When a resource joins a Private Network, the Private Network's [DHCP](/network/vpc/concepts/#dhcp) assigns it a private [IPv4](~ipv4) address (and an additional [IPv6](#ipv6) address for IPv6-compatible resources) for identification on that network. This IP is selected from the Private Network's [CIDR](/network/vpc/concepts/#cidr-block) block. The IP address will then not change unless you detach the resource from the Private Network, or delete the resource. 

Reviewer 2: ```suggestion
When a resource joins a Private Network, the Private Network's [DHCP](/network/vpc/concepts/#dhcp) assigns it a private [IPv4](#ipv4) address (and an additional [IPv6](#ipv6) address for IPv6-compatible resources) for identification on that network. This IP is selected from the Private Network's [CIDR](/network/vpc/concepts/#cidr-block) block. The IP address will then not change unless you detach the resource from the Private Network, or delete the resource. 
```

## Public (flexible) IP address

Public IP addresses identify devices on the public internet. You can enter the public IP address of an Instance into any browser connected to the Internet, and access content being served from that Instance. 

> Depending on the type of resource and its configuration, it may or may not have a public IP address. For example, you can choose to give an [Instance](/compute/instances/) one or more public IP addresses ([IPv4] and/or IPv6), or alternatively, no public IP address at all.

Reviewer 2: ```suggestion
Depending on the type of resource and its configuration, it may or may not have a public IP address. For example, you can choose to give an [Instance](/compute/instances/) one or more public IP addresses ([IPv4](#ipv4) and/or IPv6), or alternatively, no public IP address at all.
```
